internal/gatewayapi/listener.go

```diff
@@ -96,7 +96,15 @@ func (t *Translator) ProcessListeners(gateways []*GatewayContext, xdsIR XdsIRMap
 
 			// Add the listener to the Xds IR
 			servicePort := &protocolPort{protocol: listener.Protocol, port: int32(listener.Port)}
-			containerPort := servicePortToContainerPort(servicePort.port)
+
+			containerPort := servicePort.port
```

Comment:
just out of curiosity, is there any better way to improve this looong hideous nested nil check?